### PR TITLE
feat: add uat folder and codebuild workflow to run uats using otf

### PR DESCRIPTION
**Pull request by abanthiy**
**Issue #, if available:**
Want to run UATs on PRs

**Description of changes:**
Adding workflow to run UATs upon PR. 

**Why is this change necessary:**
This is to enable nucleus UAT runs on PRs using OTF.

**How was this change tested:**
- [ ] Updated or added new unit tests.
- [ ] Updated or added new integration tests.
- [x] Updated or added new end-to-end tests.
- [ ] If my code makes a remote network call, it was tested with a proxy.

**Any additional information or context required to review the change:**

**Documentation Checklist:**
 - [ ] Updated the README if applicable.

**Compatibility Checklist:**
- [ ] I confirm that the change is backwards compatible.
- [ ] Any modification or deletion of public interfaces does not impact other plugin components.
- [ ] For external library version updates, I have reviewed its change logs and Nucleus does not consume 
  any deprecated method or type.

Refer to [Compatibility Guidelines](/COMPATIBILITY.md) for more information.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/maven.yml [hunk 1]
```diff
@@ -96,3 +101,24 @@ jobs:
           name: pr
           path: pr/
         if: github.event_name == 'pull_request' && matrix.os == 'ubuntu-latest'
+
+  uat-linux:
```
Comment (MikeDombo) [inline .github/workflows/maven.yml:105]: let's move this into a separate workflow file please. And set it to run only on pull_request, not on push.

File: pom2.xml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (MikeDombo) [inline pom2.xml:1]: move to uat subdirectory?

File: uat/src/main/java/com.aws.greengrass/HelloWorld.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        if (componentName.equals("HelloWorld")) {
+            helloWorld();
+        } else if (componentName.equals("HelloWorldUpdated")) {
+            helloWorldUpdated();
+        }
+    }
+
+
+
+}
```
Comment (MikeDombo) [inline uat/src/main/java/com.aws.greengrass/HelloWorld.java:35]: EOF

File: uat/src/main/resources/greengrass/features/component.feature [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+      | com.aws.HelloWorld | classpath:/greengrass/recipes/recipe.yaml |
+    And I deploy the Greengrass deployment configuration
+    Then the Greengrass deployment is COMPLETED on the device after 180 seconds
+    And the com.aws.HelloWorld log on the device contains the line "Hello World!!" within 20 seconds
+    # Deployment with new version
+    When I create a Greengrass deployment with components
+      | com.aws.HelloWorld | classpath:/greengrass/recipes/updated_recipe.yaml |
+    And I deploy the Greengrass deployment configuration
+    Then the Greengrass deployment is COMPLETED on the device after 180 seconds
+    And the com.aws.HelloWorld log on the device contains the line "Hello World Updated!!" within 20 seconds
```
Comment (MikeDombo) [inline uat/src/main/resources/greengrass/features/component.feature:19]: EOF

File: uat/src/main/resources/greengrass/recipes/recipe.yaml [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+ComponentPublisher: Amazon
+Manifests:
+  - Artifacts:
+      - URI: "classpath:/greengrass/artifacts/componentArtifact.jar"
+        Permission:
+          Read: ALL
+          Execute: ALL
+    Lifecycle:
+      Run: |
+        java -DcomponentName="HelloWorld" -cp {artifacts:path}/componentArtifact.jar com.aws.greengrass.HelloWorld
```
Comment (MikeDombo) [inline uat/src/main/resources/greengrass/recipes/recipe.yaml:20]: EOF

File: uat/src/main/resources/greengrass/recipes/updated_recipe.yaml [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+ComponentPublisher: Amazon
+Manifests:
+  - Artifacts:
+      - URI: "classpath:/greengrass/artifacts/componentArtifact.jar"
+        Permission:
+          Read: ALL
+          Execute: ALL
+    Lifecycle:
+      Run: |
+        java -DcomponentName="HelloWorldUpdated" -cp {artifacts:path}/componentArtifact.jar com.aws.greengrass.HelloWorld
```
Comment (MikeDombo) [inline uat/src/main/resources/greengrass/recipes/updated_recipe.yaml:20]: EOF

File: pom2.xml (hunk 2 above)
Comment (urvashijain18) [reply pom2.xml:1]: We need this in root folder to generate otf-uat jar from root directory as uat directory is no under src directory. We did to keep otf-uat logic separate from nucleus.

Comment (MikeDombo) [reply pom2.xml:1]: I don't think that's right. You provide the path to this file in the codebuild definition, so you can just change that path there. If needed, you can of course copy the built artifacts somewhere else too, again using the codebuild definition

**Review by MikeDombo — APPROVED** [review]
Please update the PR title to conform to conventional commits format.

**Review by Xuqing888 — APPROVED** [review]

**Pull request merged**